### PR TITLE
Optimize imports of java modules

### PR DESCRIPTION
**Pull request by jomarko**

### CODE REVIEW

File: packages/dashbuilder/appformer/pom.xml [hunk 1]
```diff
@@ -487,19 +487,6 @@
...
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-checkstyle-plugin</artifactId>
-        <configuration>
-          <configLocation>org/uberfire/build/uberfire_checks.xml</configLocation>
-          <headerLocation>org/uberfire/build/uberfire_license_header.txt</headerLocation>
-        </configuration>
-      </plugin>
-    </plugins>
-  </reporting>
```
Comment (jomarko) [inline packages/dashbuilder/appformer/pom.xml:501]: Pointing to non existing files

File: packages/dashbuilder/checkstyle-suppressions.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ 
+<suppressions>
+  <suppress checks="FileTabCharacter"
+            files=".*\.[java,properties]"/>
+  <suppress checks="RegexpHeader"
+            files=".*\.java"/>
+  <suppress checks="NeedBraces"
+            files=".*\.java"/>
+  <suppress checks="EqualsHashCode"
+            files=".*\.java"/>
```
Comment (jomarko) [inline packages/dashbuilder/checkstyle-suppressions.xml:15]: These are corrupted in many files. Can address them in future if there is a spare time.

File: packages/dashbuilder/pom.xml [hunk 3]
```diff
@@ -92,12 +92,35 @@
...
+\* See the License for the specific language governing permissions and$\n^
+\* limitations under the License\.$\n^
+\*\/$
+]]>
+    </checkstyle.header.template>
+
+    <!-- Properties from DROOLS-WB parent POM -->
+    <!-- TODO: Enable these flags after fixing all checkstyle issues -->
+    <checkstyle.failOnViolation>true</checkstyle.failOnViolation>
+    <checkstyle.logViolationsToConsole>true</checkstyle.logViolationsToConsole>
```
Comment (jomarko) [inline packages/dashbuilder/pom.xml:123]: as `dashbuilder` does not extend kie-parent (7.x stream), needed to add this

File: packages/dashbuilder/pom.xml [hunk 4]
```diff
@@ -1119,19 +1142,6 @@
...
-              <artifactId>checkstyle</artifactId>
-              <version>${version.checkstyle}</version>
-              <exclusions>
-                <exclusion>
-                  <groupId>com.sun</groupId>
-                  <artifactId>tools</artifactId>
-                </exclusion>
-              </exclusions>
-            </dependency>
-          </dependencies>
```
Comment (jomarko) [inline packages/dashbuilder/pom.xml:1134]: alignment with `stunner-editors` checkstyle

File: packages/dashbuilder/pom.xml [hunk 5]
```diff
@@ -1165,7 +1175,62 @@
...
+                <outputFile>${project.build.directory}/checkstyle.log</outputFile>
+                <includeTestSourceDirectory>true</includeTestSourceDirectory>
+                <includeResources>true</includeResources>
+                <includeTestResources>true</includeTestResources>
+                <consoleOutput>false</consoleOutput>
+                <logViolationsToConsole>${checkstyle.logViolationsToConsole}</logViolationsToConsole>
+                <failsOnError>false</failsOnError>
+              </configuration>
+            </execution>
+          </executions>
```
Comment (jomarko) [inline packages/dashbuilder/pom.xml:1233]: as dashbuilder does not extend kie-parent (7.x stream), needed to add this, probably we should create `checkstyle-maven-bom` in future to have single location for checkstyle configuration? What do you think?

File: packages/stunner-editors/checkstyle-suppressions.xml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+<suppressions>
+  <suppress checks="FileTabCharacter"
+            files=".*\.[java,properties]"/>
+  <suppress checks="RegexpHeader"
+            files=".*\.java"/>
+  <suppress checks="NeedBraces"
+            files=".*\.java"/>
+  <suppress checks="EqualsHashCode"
+            files=".*\.java"/>
+</suppressions>
```
Comment (jomarko) [inline packages/stunner-editors/checkstyle-suppressions.xml:16]: these rules are corrcupted in many files, can address it in future if there is a spare time.

**Review by jomarko — COMMENTED** [review]
Some explanation to config changes. Let me know your opinion please.

**Comment by tiagobento** [comment]
@jomarko Can you please rebase this PR? The Ci had a problem earlier today.

**Comment by jomarko** [comment]
@tiagobento rebased

File: packages/dashbuilder/pom.xml (hunk 5 above)
Comment (yesamer) [reply packages/dashbuilder/pom.xml:1233]: @jomarko There is a planned task to refactor the parent module of `kie-tools` here: https://issues.redhat.com/browse/KOGITO-6378. I would consider to manage it in that ticket.

**Review by yesamer — APPROVED** [review]
@jomarko Thank you for this huge work! Priceless! I check only a subset of the files your changes, considering the approach you used seems correct to me and the green lights we have, I think we can safely proceed to merge it.

**Comment by jomarko** [comment]
@tiagobento @romartin hello, just a friendly reminder if you have a time to review this PR

**Review by romartin — APPROVED** [review]
Thanks @jomarko .
I don't see anything potentially conflicting, although it's a huge PR, so please just be sure webapps are working (even in SDM) after this change. :+1:

**Comment by jomarko** [comment]
@yesamer can I get a merge please?

**Pull request merged**